data_steward/utils/bq.py

```diff
@@ -240,3 +242,70 @@ def get_latest_validation_dataset_id(project_id):
             return sorted(validation_datasets, key=lambda x: x[0],
                           reverse=True)[0][1]
     return None
+
+
+def dataset_exists(dataset_id, project_id):
+    """
+    Checks if the dataset exists via dataset id
+
+    :param dataset_id:  ID (name) of the dataset to check
+    :param project_id:  ID (name) of the project to check for the dataset
+
+    :raises:  RuntimeError if a project_id is not supplied
+    :raises:  RuntimeError if a dataset_id is not supplied
+    :returns:  True if the dataset exists
+    :returns:  False if the dataset does not exist
+    """
+    if not project_id:
+        raise RuntimeError("Please specify a project")
+    if not dataset_id:
+        raise RuntimeError("Please specify a dataset to check if exists")
+    else:
+        client = get_client(project_id)
+        dataset_id = f"{project_id}.{dataset_id}"
+        dataset = bigquery.Dataset(dataset_id)
+
+        try:
+            client.get_dataset(dataset)
+            return True
+        except NotFound:
+            return False
+
+
+def create_dataset(dataset_id, description, label_or_tag, project_id):
+    """
+    Creates a new dataset
+
+    :param project_id:  ID (name) of the project in which to create the dataset, is required
+    :param dataset_id:  ID (name) to give the new dataset, is required
+    :param description:  description of the dataset, is required
+    :param label_or_tag:  Dict[str,str] labels or tag for the dataset
+
+    :raises: RuntimeError if the dataset does not have project_id
+    :raises: RuntimeError if the dataset does not have dataset_id
+    :raises: RuntimeError if the dataset does not have a description
+    :raises: RuntimeError if the dataset does not have a label
+    """
+    if dataset_exists(dataset_id, project_id) == False:
+        if description.isspace() or not description:
+            raise RuntimeError(
+                "Please provide a description to create a dataset")
+
+        if not label_or_tag:
```

Comment:
when the label and tag are separated. I think it is better to check if either one of them both is specified. Not all the datasets might have both labels and tags.